MoodDetectorWebApp/Controllers/ManageUsersController.cs

```diff
@@ -49,16 +50,42 @@ public ActionResult AddUser(AddUserModel addUserModel)
             return View("SuccessfulAdd");
         }
 
-        // GET: EditUser
-        public ActionResult EditUser()
+        [HttpGet]
+        public ActionResult EditUser(int id)
         {
-            return View();
+            var user = _userService.GetAddUser(id);
+            return View("EditUser", user);
         }
 
-        // GET: DeleteUser
-        public ActionResult DeleteUser()
+        [HttpPost]
+        public ActionResult EditUser(AddUserModel addUserModel, int id)
         {
-            return View();
+            AddUser addUser = new AddUser(
+                addUserModel.Username,
+                addUserModel.Password,
+                addUserModel.Email,
+                addUserModel.Firstname,
+                addUserModel.Lastname,
+                addUserModel.AccessRights
+                );
+            /*if (!_registerAuthenticator.IsRegisterDataCorrect(addUser))
+            {
+                return View("DeleteUser");
+            }*/
+            _userService.EditUser(addUser, id);
```

Comment:
Keistai addUser, kai editinamas user'is. Čia gali naudoti User pilnai tiks.

Reply:
bet gali būti editinamas ne tik useris, bet ir jo loginInfo

Reply:
Supratau. Tai galbūt naming reikia pakeisti, kadangi struct naudojamas plačiau nei jo pavadinimas parodo.